### ISSUE TITLE
2.0.0-M.5 - Issue with custom XmlAdapter for enums

### ISSUE THREAD

**Issue by scoquelin** [stoicflame/enunciate#147]
This issue might be a bit to specific but reporting it anyway in case there could be a quick fix.
Reason why we did that : 
http://stackoverflow.com/questions/12147306/handling-invalid-enum-values-while-doing-jaxb-unmarshalling

Now let's assume I have an enum defined as : 

````
@XmlJavaTypeAdapter(MyEnumAdapter.class)
public enum MyEnum {
...
}
````

and adapter defined such as : 

````
public class MyEnumAdapter extends EnumAdapter<MyEnum> {
...
}
````

where we have : 

````
public abstract class EnumAdapter<T extends Enum<T>> extends XmlAdapter<String, T> {
...
}
````

Then I get the following error : 

**java.lang.RuntimeException: com.webcohesion.enunciate.EnunciateException: null: adapter com.myproject.MyEnumAdapter does not adapt com.myproject.MyEnum**

Complete stacktrace : 
````
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.5:assemble (assemble) on project myproject: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.5:assemble failed: com.webcohesion.enunciate.EnunciateException: null: adapter com.myproject.MyEnumAdapter does not adapt com.myproject.MyEnum: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.5:assemble (assemble) on project myproject: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.5:assemble failed: com.webcohesion.enunciate.EnunciateException: null: adapter com.myproject.MyEnumAdapter does not adapt com.myproject.MyEnum
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.5:assemble failed: com.webcohesion.enunciate.EnunciateException: null: adapter com.myproject.MyEnumAdapter does not adapt com.myproject.MyEnum
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:143)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: java.lang.RuntimeException: com.webcohesion.enunciate.EnunciateException: null: adapter com.myproject.MyEnumAdapter does not adapt com.myproject.MyEnum
	at com.sun.tools.javac.main.Main.compile(Main.java:553)
	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
	at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:650)
	at com.webcohesion.enunciate.mojo.ConfigMojo.execute(ConfigMojo.java:356)
	at com.webcohesion.enunciate.mojo.AssembleBaseMojo.execute(AssembleBaseMojo.java:86)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	... 20 more
Caused by: com.webcohesion.enunciate.EnunciateException: null: adapter com.myproject.MyEnumAdapter does not adapt com.myproject.MyEnum
	at com.webcohesion.enunciate.modules.jaxb.model.util.JAXBUtil.findAdapterType(JAXBUtil.java:190)
	at com.webcohesion.enunciate.modules.jaxb.model.util.JAXBUtil.findAdapterType(JAXBUtil.java:152)
	at com.webcohesion.enunciate.modules.jaxb.model.util.MapType.findMapTypeDeclaration(MapType.java:133)
	at com.webcohesion.enunciate.modules.jaxb.model.util.MapType.findMapType(MapType.java:86)
	at com.webcohesion.enunciate.modules.jaxb.model.Accessor.getAccessorType(Accessor.java:126)
	at com.webcohesion.enunciate.modules.jaxb.model.Element.getAccessorType(Element.java:211)
	at com.webcohesion.enunciate.modules.jaxb.model.util.JAXBUtil.findAdapterType(JAXBUtil.java:144)
	at com.webcohesion.enunciate.modules.jaxb.model.Accessor.<init>(Accessor.java:73)
	at com.webcohesion.enunciate.modules.jaxb.model.Element.<init>(Element.java:51)
	at com.webcohesion.enunciate.modules.jaxb.model.TypeDefinition.<init>(TypeDefinition.java:137)
	at com.webcohesion.enunciate.modules.jaxb.model.SimpleTypeDefinition.<init>(SimpleTypeDefinition.java:33)
	at com.webcohesion.enunciate.modules.jaxb.model.ComplexTypeDefinition.<init>(ComplexTypeDefinition.java:34)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.createTypeDefinition(EnunciateJaxbContext.java:281)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:836)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:801)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:699)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:635)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:375)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:836)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:801)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:699)
	at com.webcohesion.enunciate.modules.jaxb.JaxbModule.addDataTypeDefinitions(JaxbModule.java:78)
	at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.addReferencedDataTypeDefinitions(JaxrsModule.java:164)
	at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.call(JaxrsModule.java:102)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
	... 23 more````

Thanks!

**Comment by stoicflame** [comment]
Fixed at b4b7a96.

**Comment by stoicflame** [comment]
Please confirm the fix with [Release Candidate 1](https://github.com/stoicflame/enunciate/releases/tag/v2.0.0-RC.1)
